### PR TITLE
Changes to SampleMetadata config class.

### PR DESCRIPTION
**Pull request by milo-hyben**
Before we can deploy the latest Metamist ETL, we would need to add extra config to SampleMetadata class, here are my suggestions.
All are prefixed as ETL_

### CODE REVIEW

File: cpg_infra/config/config.py [hunk 1]
```diff
@@ -123,6 +123,10 @@ class GCP(DeserializableDataclass):
         gcp: GCP
         slack_channel: str | None = None
         etl_accessors: list[str] = dataclasses.field(default_factory=list)
+        etl_environment: str | None = None
+        etl_default_config: dict[str, str] | None = None
```
Comment (illusional) [inline cpg_infra/config/config.py:127]: Can you add a couple of comments describing what values you'd expect in these fields?

File: cpg_infra/config/config.py [hunk 2]
```diff
@@ -123,6 +123,10 @@ class GCP(DeserializableDataclass):
...
+        etl_private_repo_url: str | None = None
```
Comment (illusional) [inline cpg_infra/config/config.py:128]: The private_repo_url you should be able to pull directly from the resource created in #152 (I hope)

```suggestion
```

File: cpg_infra/config/config.py [hunk 3]
```diff
@@ -123,6 +123,10 @@ class GCP(DeserializableDataclass):
...
+        etl_private_repo_packages: str | None = None
```
Comment (illusional) [inline cpg_infra/config/config.py:129]: Is this a comma separated list of packages to install, what would you expect here?

File: cpg_infra/config/config.py (hunk 1 above)
Comment (milo-hyben) [reply cpg_infra/config/config.py]: Comments and example values added, I had renamed config to 'etl_parser_default_config' to make the name clearer.

File: cpg_infra/config/config.py (hunk 2 above)
Comment (milo-hyben) [reply cpg_infra/config/config.py]: Good point, I had removed that entry.

File: cpg_infra/config/config.py (hunk 3 above)
Comment (milo-hyben) [reply cpg_infra/config/config.py]: Change the packages to a list, more clearer this way.

File: cpg_infra/config/config.py [hunk 4]
```diff
@@ -123,6 +123,13 @@ class GCP(DeserializableDataclass):
         gcp: GCP
         slack_channel: str | None = None
         etl_accessors: list[str] = dataclasses.field(default_factory=list)
+        # Metamist environment (DEVELOPMENT / PRODUCTION) for ETL cloud functions
+        etl_environment: str | None = None
```
Comment (illusional) [inline cpg_infra/config/config.py:127]: Can you provide 'PRODUCTION' as an explicit default here?

**Review by illusional — APPROVED** [review]

File: cpg_infra/config/config.py (hunk 4 above)
Comment (milo-hyben) [reply cpg_infra/config/config.py]: No worries, changed that default.

**Pull request merged**